### ISSUE TITLE
Should Burnham use the client activity API?

### ISSUE THREAD

**Issue by badboy** [mozilla/burnham#148]
**Is your feature request related to a problem? Please describe.**

Glean now provides the client activity API.
It is [not yet implemented for Python](https://bugzilla.mozilla.org/show_bug.cgi?id=1688842).
This allows embedding applications to trigger behavior based on client activity.
On mobile this maps to foregrounding (active) and backgrounding (inactive).
For Desktop this maps to certain activity within a certain amount of time.

The API triggers the baseline and events ping, both otherwise internal pings.

**Describe the solution you'd like**

Would it be useful for burnham to use this API to simulate activity and thus test Glean's expected behavior around those pings?

**Comment by hackebrot** [comment]
Yes, I think adding meaningful test coverage for the client activity API would be useful! 👍🏻

**Comment by badboy** [comment]
Ok, let's design one! (As said, the implementation is not yet done, but should be available in the next release then)

**Comment by hackebrot** [comment]
Status update: @badboy will bring this to the team's attention today. 🚧

**Comment by badboy** [comment]
We currently don't have the time to pick up this work.

**Comment by hackebrot** [comment]
Hi @chutten! 👋🏻 

From what I understand FOG is one of the consumers of the activity API. Do you have any thoughts or suggestions for how we can meaningfully test this API with burnham?

**Comment by chutten** [comment]
Yup, FOG certainly is using the client activity API to signal activity (which Glean turns into "baseline" pings, as well as an opportunity to send "events" pings). Burnham could use these APIs to ensure pings are sent with the appropriate reasons at the appropriate times (and in the appropriate numbers) by simulating normal use (actives followed by inactives followed by actives) and error cases (what happens if we call active twice in a row? inactive? What if the first API call is to inactive instead of active?)

This would be important for understanding the failure modes of the client activity API and how visible they are in the data. Maybe FOG's misusing the API in a way that is presently invisible?

**Comment by hackebrot** [comment]
Thank you for adding your input, @chutten! 👍🏻
